### PR TITLE
refactor: make all the utils TS

### PR DESCRIPTION
**Pull request by vladfrangu**
Will depend on https://github.com/apify/apify-shared-js/issues/131

CC: @pocesar @B4nan @mnmkng

### CODE REVIEW

**Comment by pocesar** [comment]
@vladfrangu there's #24 most of it is done

**Comment by vladfrangu** [comment]
yeah, I'm planning to for the bigger classes since I saw you did a *lot* with generics, unless you meant something else! 😋

**Comment by mnmkng** [comment]
@B4nan could you please check this?

@vladfrangu sorry for the long delay, https://github.com/apify/apify-shared-js/issues/131 is closed now and the shared repo is fully in TS, you can continue with the work.

**Review by B4nan — APPROVED** [review]
Other than the logger issue it looks fine

File: src/logger.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import defaultLog from "@apify/log";
```
Comment (B4nan) [inline src/logger.ts:1]: single quotes

File: src/logger.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore For now till the logger becomes its own module
+export const log = defaultLog.child({
+    prefix: "BrowserPool",
```
Comment (B4nan) [inline src/logger.ts:6]: single quotes

**Review by B4nan — CHANGES_REQUESTED** [review]
nit pick

**Comment by vladfrangu** [comment]
@B4nan Should I disable the `import/extensions` rule globally? It's not needed at all in JS/TS, and from one of your commits you even removed the extension from some imports 👀

**Comment by B4nan** [comment]
afaik in TS we should not use extensions in imports, or something changed? :] i know they are required to use ESM, but we are not at that boat just yet and i expect TS to handle that for us anyway 

but i dont get the context, do we have some extensions in imports in the repo?

**Comment by vladfrangu** [comment]
> but i dont get the context, do we have some extensions in imports in the repo?

image

**Comment by vladfrangu** [comment]
Alrighty! Then this PR is review-and-merge ready 👍

**Review by B4nan — APPROVED** [review]

**Comment by B4nan** [comment]
Looking at that commit I made and the linter passed just fine on that, weird. Maybe we should really disable that rule, personally I am not a big fan of airbnb style guide anyway :D 

Or fix it now in the JS files, but the extensions should not be there once its TS.

**Comment by vladfrangu** [comment]
Scratch that, turns out the errors are caused by the fact the files are now `.ts` instead of `.js`, which the rule dislikes... 🙃 
And we cannot import .ts files, since they don't exist at runtime... Should I just disable the rule for now in the module instead?

**Comment by B4nan** [comment]
yeah just add ignore comments i guess, we can deal with that later.

**Pull request merged**